[ko] queryLocalFonts 신규 번역 (#25655)
mdn/translated-content: files/ko/web/api/window/querylocalfonts/index.md

---
title: "Window: queryLocalFonts() method"
slug: Web/API/Window/queryLocalFonts
status:
  - experimental
l10n:
  sourceCommit: d7143e171b5f18fb37a686a7d4947db417fd74f3
---

{{APIRef("Local Font Access API")}}{{SeeCompatTable}}{{SecureContext_Header}}

**`window.queryLocalFonts()`** 메서드는 로컬에서 사용 가능한 폰트 페이스를 나타내는 {{domxref("FontData")}} 객체의 배열을 충족하는 {{jsxref("Promise")}} 를 반환합니다.

이 메서드를 사용하기 위해서 사용자는 반드시 `local-fonts` ({{domxref("Permissions API", "", "", "nocode")}}를 통한 권한 상태) 접근 권한을 허용해야 합니다. 또한, 이 특성은 사용자의 서버에 설정된 [권한 정책](/ko/docs/Web/HTTP/Permissions_Policy)에 의하여 차단될 수 있습니다.

## 구문

```js-nolint
queryLocalFonts(options)
```

> ### 파라미터

Reviewer: ```suggestion
### 매개변수
```

문서의 통일성을 위해 용어를 맞추고 있습니다.

용어집은 [Glossary guide](https://github.com/mdn/translated-content/blob/main/docs/ko/guides/glossary-guide.md#section-title)에서 확인할 수 있습니다.

아래 스니펫은 사용 가능한 모든 폰트를 조회하고 메타데이터를 기록합니다. 예를 들어 폰트 선택 제어를 채우는 데에 사용할 수 있습니다.

```js
async function logFontData() {
  try {
    const availableFonts = await window.queryLocalFonts();
    for (const fontData of availableFonts) {
      console.log(fontData.postscriptName);
      console.log(fontData.fullName);
      console.log(fontData.family);
      console.log(fontData.style);
    }
  } catch (err) {
    console.error(err.name, err.message);
  }
}
```

### 반환된 결과 제어하기

`postscriptNames` 기능을 사용하면 반환된 폰트 데이터를 폰트 페이스의 특정 리스트로 제한할 수 있습니다.

```js
async function returnSpecificFonts() {
  const availableFonts = await window.queryLocalFonts({
    postscriptNames: ["Verdana", "Verdana-Bold", "Verdana-Italic"],
  });

  return availableFonts;
}
```

### 낮은 레벨의 데이터에 접근하기

{{domxref("FontData.blob", "blob()")}} 메서드는 낮은 레벨의 [SFNT](https://en.wikipedia.org/wiki/SFNT) 데이터에 접근할 수 있도록 합니다. SFNT는 PostScript, TrueType, OpenType, Web Open Font Format(WOFF) 과 같은 다른 폰트 형식을 포함할 수 있는 폰트 파일 형식입니다.

```js
async function computeOutlineFormat() {
  try {
    const availableFonts = await window.queryLocalFonts({
      postscriptNames: ["ComicSansMS"],
    });
    for (const fontData of availableFonts) {

>       // `blob()` returns a Blob containing valid and complete
>       // SFNT-wrapped font data.

Reviewer: ```suggestion
      // `blob()`은 유효하고 완전한 SFNT 래핑된 폰트 데이터를 포함하는
      // Blob을 반환합니다.
```

      const sfnt = await fontData.blob();

>       // Slice out only the bytes we need: the first 4 bytes are the SFNT
>       // version info.
>       // Spec: https://learn.microsoft.com/ko/typography/opentype/spec/otff#organization-of-an-opentype-font

Reviewer: ```suggestion
      // 필요한 바이트만 잘라냅니다. 처음 4바이트는 SFNT
      // 버전 정보입니다.
      // 명세: https://learn.microsoft.com/ko/typography/opentype/spec/otff#organization-of-an-opentype-font
```
코드 주석이 아닌 경우 함께 번역되면 좋을거 같아요.

      const sfntVersion = await sfnt.slice(0, 4).text();

      let outlineFormat = "UNKNOWN";
      switch (sfntVersion) {
        case "\x00\x01\x00\x00":
        case "true":
        case "typ1":
          outlineFormat = "truetype";
          break;
        case "OTTO":
          outlineFormat = "cff";
          break;
      }
      console.log("Outline format:", outlineFormat);
    }
  } catch (err) {
    console.error(err.name, err.message);
  }
}
```

## 명세서

{{Specifications}}

## 브라우저 호환성

{{Compat}}

## 같이 보기

- {{domxref("Local Font Access API", "Local Font Access API", "", "nocode")}}
- [로컬 폰트와 함께 향상된 타이포그래피 사용하기](https://developer.chrome.com/docs/capabilities/web-apis/local-fonts)

> - {{cssxref("@font-face")}}[권한 정책](/ko/docs/Web/HTTP/Permissions_Policy)

Reviewer: ```suggestion
- {{cssxref("@font-face")}}
```

잘못 추가된 것 같아요.
